terraform/environments/digital-prison-reporting/main.tf

```diff
@@ -1068,6 +1068,61 @@ module "kinesis_cdc_domain_stream" {
   cloudwatch_logging_enabled = true
 }
 
+# Dynamo DB for DomainRegistry, DPR-306/DPR-218
+module "dynamo_tab_domain_registry" {
+  source              = "./modules/dynamo_tables"
+  create_table        = true
+  autoscaling_enabled = false
+  name                = "${local.project}-domain-registry-${local.environment}"
+
+  hash_key    = "primaryId"
+  range_key   = "secondaryId"
+  table_class = "STANDARD"
+  ttl_enabled = false
+
+  attributes = [
+    {
+      name = "primaryId"
+      type = "S"
+    },
+    {
+      name = "secondaryId"
+      type = "S"
+    },
+    {
+      name = "type"
+      type = "S"
+    }
```

Comment:
There is a fourth attribute on the table `data` which is type `S`.

Reply:
So this is interesting, what I am seeing is - if we add a attribute without a key, I see below error

all attributes must be indexed. Unused attributes: [“"]

what I understood since it is schema less, we should add the indexed attributes and let the application create other attributes as required ?

Reply:
Hmm...ok. I wasn't aware that we could add fields on the fly since that's not how I've been using it. We don't need to index the data field since in normal usage we won't be searching the contents of this field. I'll have a little look into this and get back to you.

Reply:
So googling around it turns out that the attribute blocks are used to define the key schema for the table so we're good here. Any additional fields that aren't going to be indexed do not need to be specified in this section.

More info here https://stackoverflow.com/questions/50006885/terraform-dynamodb-all-attributes-must-be-indexed